### ISSUE TITLE
I can't download from Youtube !

### ISSUE THREAD

**Issue by clodez** [persepolisdm/persepolis#497]
**System Details:**
* Operating System: Unix
* Distro(For GNU/Linux and BSD users): Freebsd 11.1
* Desktop Environment(For GNU/Linux and BSD users): XFCE
* Persepolis Version: 3.1
* How do you install Persepolis? (Repositories, source, ...)
Source latest
* _Please attach log files if crash or paste error message_
Persepolis Download Manager - ERROR - Download failed - GID : e03f40dafeff530c/nMessage : SSL/TLS handshake failure: unable to get local issuer certificate

**Issue Description and steps to reproduce:**
I tried all possible tricks for aria2.conf to make this work like creating the file and adding 
ca-certificate=/usr/local/share/certs/ca-root-nss.crt
note is that I have installed youtube_dl using pip with => sudo pip-3.6 install --upgrade youtube_dl 

awaiting Your feedback .

**Comment by alireza-amirsamimi** [comment]
Hello
Persepolis dosn't use user's conf file.

Question: can persepolis find the link with VideoFinder?

**Comment by clodez** [comment]
yes it can , and it can detect several video resolution from the drop down menu button .

**Comment by alireza-amirsamimi** [comment]
I have no problem with youtube, ... links. It's related to aria2c.
Do you have same problem with [DailyMotion](http://www.dailymotion.com/video/x46eaz) or [Vimeo](https://vimeo.com/246959904) ?

**Comment by clodez** [comment]
seems You are right I have a Problem with all https links ! I can download http just fine ex : http://download.videolan.org/pub/videolan/vlc/3.0.1/vlc-3.0.1.tar.xz

**Comment by alireza-amirsamimi** [comment]
@clodez Try dns resolvers or Tor. May they fix the problem.

**Comment by clodez** [comment]
please explain more about how to try Tor or dns resolver !!

**Comment by alireza-amirsamimi** [comment]
@clodez Unfortunately I'm not familar with BSD :(
May this [link](https://www.torproject.org/) helps you.
If You use tor or tor browser, You need also [privoxy](https://github.com/persepolisdm/persepolis/wiki/Privoxy).

**Comment by clodez** [comment]
I don't think it's a browser issue , but it should be an aria2c SSL/TLS config or missing dependency issue , using another browser and some proxy will not fix the issue , I guess ! any other suggestions ?

**Comment by alireza-amirsamimi** [comment]
@clodez What aria2c version is installed on your system?

**Comment by clodez** [comment]
aria2 version 1.33.1

**Comment by alireza-amirsamimi** [comment]
@clodez I have this version too :) I have no idea why it happens for you.
Try Tor and privoxy :)

**Comment by clodez** [comment]
I am trying , still not able to make persepolis work with tor-proxy , yet ...

**Comment by alireza-amirsamimi** [comment]
@clodez How do you set proxy in Persepolis

**Comment by clodez** [comment]
I tried privoxy which works with firefox browser , but did not work with persepolis , when creating a new download from the GUI select the second tab Proxy .

**Comment by alireza-amirsamimi** [comment]
@clodez Do you use
127.0.0.1 for ip and 8118 for port?

**Comment by clodez** [comment]
sorry missed adding the error from persepolis log is : Download failed - GID : 947ea9802b4fa964/nMessage : SSL/TLS handshake failure: unable to get local issuer certificate , same as direct connection !

**Comment by clodez** [comment]
yes this is exactly what I am using 127.0.0.1 and port 8118

**Comment by Monirzadeh** [comment]
aria2c can download from any other https link?

**Comment by clodez** [comment]
how can I test aria2c from the command line , I tried but failed

**Comment by Monirzadeh** [comment]
try any other links that start with https, with or without proxy.
maybe the problem is similar this issue:
https://github.com/aria2/aria2/issues/781
follow the suggestion and especially test this
https://github.com/aria2/aria2/issues/781#issuecomment-262553754
for sure source of the problem is which of them Persepolis or aria2c.
`aria2c "YourlinkURL"`

**Comment by clodez** [comment]
1- 
aria2c https://download.anydesk.com/AnyDesk.exe
04/14 16:14:08 [NOTICE] Downloading 1 item(s)
04/14 16:14:08 [ERROR] Failed to load trusted CA certificates from no. Cause: error:02001002:system library:fopen:No such file or directory
04/14 16:14:08 [ERROR] CUID#7 - Download aborted. URI=https://download.anydesk.com/AnyDesk.exe

2- 
aria2c --check-certificate=false https://download.anydesk.com/AnyDesk.exe

04/14 16:15:15 [NOTICE] Downloading 1 item(s)

04/14 16:15:15 [ERROR] Failed to load trusted CA certificates from no. Cause: error:02001002:system library:fopen:No such file or directory
[#ba4b1b 1.6MiB/1.9MiB(87%) CN:1 DL:307KiB]                                                                                                                                                                                           
04/14 16:15:21 [NOTICE] Download complete: /down/AnyDesk.exe

using --check-certificate=false , makes it work , now how can I apply that to persepolis ?

**Comment by Monirzadeh** [comment]
@alireza-amirsamimi  review above report I am not sure about security that deactivates check certificate.
what do you think about that?

**Comment by ghost** [comment]
@clodez : according to [this](https://forums.freebsd.org/threads/error-with-aria2c.51545/) try installing `security/ca_root_nss` package. see the post in freebsd forum.

**Comment by clodez** [comment]
done that before posting for help , that did not work for persepolis , but I just double checked it's working from the command line !!

**Comment by Monirzadeh** [comment]
@clodez I confused!!!
https://github.com/persepolisdm/persepolis/issues/497#issuecomment-381328612 
I am not expected to work on Pespolise because Persepolis use aria2c and --check-certificate=true by default
but this test shows the problem is not from any other option

**Comment by clodez** [comment]
I am sorry , let me add what I changed in the last note that @morealaz cleared ,
I created ~/.aria2/aria2.conf and added ca-certificate=/etc/ssl/cert.pem to the conf file .
but that made aria2c work from the command line without the need to add "--check-certificate=false"
but it still fail from the GUI .

**Comment by clodez** [comment]
Guys I read a bit the code of download.py and managed to make it work by adding :  --ca-certificate=/etc/ssl/cert.pem =>

# start aria2 with RPC
def startAria():
    # in Linux and BSD
    if os_type == 'Linux' or os_type == 'FreeBSD' or os_type == 'OpenBSD':
        os.system("aria2c --version 1> /dev/null")
        os.system("aria2c --ca-certificate=/etc/ssl/cert.pem --no-conf  --enable-rpc --rpc-listen-port '" 

I am guessing --no-conf is stopping aria2c from reading the conf that I added earlier , can someone explain why are you guys using --no-conf ! . there is definitely a better way of fixing this in the code , any suggestions ?

**Comment by alireza-amirsamimi** [comment]
Persepolis configures aria itself. User can select some options in preference window. So Persepolis must call aria2 with `--no-conf` . Perhaps user's conf. file has conflict with Persepolis's configuration.

**Comment by Monirzadeh** [comment]
I am not sure but base on my knowledge Perspolis save configure for each link separately.
@alireza-amirsamimi  mention here https://github.com/persepolisdm/persepolis/issues/497#issuecomment-381317311

**Comment by terry-toerag** [comment]
I use Mandriva Scion Linux...and use youtube-dl {URL] to download youtube vids
EG youtube-dl https://www.youtube.com/watch?v=lJxVnGekbWo.
